source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -138,9 +138,17 @@ FilterStats FilterConfig::generateStats(const std::string& prefix, Stats::Scope&
 
 void OAuth2CookieValidator::setParams(const Http::RequestHeaderMap& headers,
                                       const std::string& secret) {
-  expires_ = Http::Utility::parseCookieValue(headers, "OauthExpires");
-  token_ = Http::Utility::parseCookieValue(headers, "BearerToken");
-  hmac_ = Http::Utility::parseCookieValue(headers, "OauthHMAC");
+  const auto& cookies = Http::Utility::parseCookies(headers);
```

Comment:
This implementation parses all cookies into a map even when we just need a subset.

Should `parseCookies(...)` be enhanced to accept a `filter` to filter down on keys at the time of parsing ?

Reply:
Not sure if possible performance gain can balance increased code complexity. I'd rather keep it as it is for now.